mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -432,6 +503,22 @@ object GeneralizedLinearRegression extends DefaultParamsReadable[GeneralizedLine
 
     override def variance(mu: Double): Double = math.pow(mu, 2.0)
 
+    override def deviance(y: Double, mu: Double, weight: Double): Double = {
+      val x = if (y == 0.0) 1.0 else y / mu
```

Comment:
When would `y == 0.0` happen? If this is not feasible in Gamma family, we should throw an error. In the current setting, this method returns `-2.0 * weight` when `y == 0.0`.